lib/std/mem.zig

```diff
@@ -947,6 +947,41 @@ test "len" {
     try testing.expect(len(c_ptr) == 2);
 }
 
+pub const LexicographicalOrder = enum(i2) {
+    lt = -1,
+    eq = 0,
+    gt = 1,
+
+    pub fn toMathOrder(self: LexicographicalOrder) math.Order {
+        return switch (self) {
+            .lt => .lt,
+            .eq => .eq,
+            .gt => .gt,
+        };
+    }
+};
+
+/// Returns `LexicographicalOrder.lt` if `a` appears before `b` in lexicographical order.
+/// Returns `LexicographicalOrder.gt` if `a` appears after `b` in lexicographical order.
+/// Returns `LexicographicalOrder.eq` if `a` and `b` are equal.
+/// See also `std.mem.order`.
+pub fn stringCompareZ(a: [*:0]const u8, b: [*:0]const u8) LexicographicalOrder {
+    @setRuntimeSafety(false);
+    var index: usize = 0;
+    while (a[index] == b[index] and a[index] != 0) : (index += 1) {}
+    const result: i2 = @as(i2, @boolToInt(a[index] > b[index])) - @as(i2, @boolToInt(a[index] < b[index]));
+    return @intToEnum(LexicographicalOrder, result);
```

Comment:
Please write this without using `@intToEnum` unless you can demonstrate that it affects performance.

Reply:
> Please write this without using `@intToEnum` unless you can demonstrate that it affects performance.

Well, I compared this version with `normal` version a.k.a `return math.Order` and on my PC this version was consistently slightly faster after 1_000_000 iterations (I thought due to extra branches and stuff), but right now I'm observing that normal version is much faster on my notebook (???). Looks like it's not worth to oversmart compiler and leave it with its smart optimizations. My big thanks for the reviews!